<pre><code>library(tidyverse)</code></pre>
<h1 id="clinical-datasets">Clinical datasets</h1>
<pre><code>cm &lt;- read_table(&quot;../clinicaldata/cm_clinical_data.tsv&quot;)
tz &lt;- read_table(&quot;../clinicaldata/tz_clinical_data.tsv&quot;)</code></pre>
<h2 id="summary-of-cameroon-data">Summary of Cameroon data</h2>
<h3 id="variables">variables</h3>
<pre><code>cm %&gt;% names()

##  [1] &quot;FID&quot;                 &quot;IID&quot;                 &quot;AGE&quot;                
##  [4] &quot;SEX&quot;                 &quot;HbF&quot;                 &quot;HbS&quot;                
##  [7] &quot;Hb_Phenotype&quot;        &quot;Hb&quot;                  &quot;Phenotypes&quot;         
## [10] &quot;Genotypes&quot;           &quot;Haplotypes&quot;          &quot;athal&quot;              
## [13] &quot;Curated_Age_yrs&quot;     &quot;VOC.year&quot;            &quot;Consultation.year&quot;  
## [16] &quot;Hosp&quot;                &quot;Stroke&quot;              &quot;HbA2_percentage&quot;    
## [19] &quot;RBC&quot;                 &quot;MCV.fl.&quot;             &quot;MCHC&quot;               
## [22] &quot;WBC&quot;                 &quot;Lymp&quot;                &quot;Mono&quot;               
## [25] &quot;Platelets&quot;           &quot;Curated_penotype&quot;    &quot;Sample_name_comment&quot;</code></pre>
<h3 id="summary-statistics">Summary statistics</h3>
<p><strong>Quantitative/Continuous variables</strong></p>
<pre><code>variables &lt;- c(
   &quot;AGE&quot;,                   # Age in years
   &quot;HbS&quot;,                   # Proportion of HbS
   &quot;HbF&quot;,                   # Proportion of fetal haemoglobin
   &quot;HbA2_percentage&quot;,       # Proportion of HbA2
   &quot;Consultation.year&quot;,     # Number of consultations per year
   &quot;MCHC&quot;,                  # Mean corpuscular haemoglobin concentration
   &quot;Mono&quot;,                  # Monocyte count (x10^9/l)
   &quot;Hosp&quot;,                  # Number of hospitalizations per year
   &quot;RBC&quot;,                   # Red blood cell count (x10^12/l)
   &quot;WBC&quot;,                   # White blood cell count (x10^9/l)
   &quot;Platelets&quot;,             # Platelet count (x10^9/l)
   &quot;VOC.year&quot;,              # Number of vaso occlusive crises per year
   &quot;MCV.fl.&quot;,               # Mean cell volume
   &quot;Lymp&quot;                   # Lymphocyte count (x10^9/l)
)
cm %&gt;% select(variables) %&gt;% summary()

##       AGE             HbS             HbF         HbA2_percentage 
##  Min.   : 5.00   Min.   :19.60   Min.   : 0.700   Min.   : 0.000  
##  1st Qu.:10.00   1st Qu.:80.30   1st Qu.: 4.700   1st Qu.: 2.700  
##  Median :16.00   Median :86.00   Median : 8.500   Median : 3.200  
##  Mean   :18.05   Mean   :83.56   Mean   : 9.892   Mean   : 3.324  
##  3rd Qu.:24.00   3rd Qu.:90.70   3rd Qu.:13.600   3rd Qu.: 3.800  
##  Max.   :66.00   Max.   :96.50   Max.   :37.400   Max.   :18.200  
##                                                   NA&#39;s   :3       
##  Consultation.year      MCHC            Mono            Hosp       
##  Min.   : 0.000    Min.   : 3.80   Min.   :0.000   Min.   : 0.000  
##  1st Qu.: 0.000    1st Qu.:25.90   1st Qu.:0.700   1st Qu.: 1.000  
##  Median : 1.000    Median :31.00   Median :1.000   Median : 1.000  
##  Mean   : 2.049    Mean   :29.21   Mean   :1.161   Mean   : 1.974  
##  3rd Qu.: 2.000    3rd Qu.:33.00   3rd Qu.:1.400   3rd Qu.: 2.000  
##  Max.   :15.000    Max.   :54.20   Max.   :9.400   Max.   :24.000  
##  NA&#39;s   :805       NA&#39;s   :17      NA&#39;s   :14      NA&#39;s   :59      
##       RBC            WBC           Platelets         VOC.year     
##  Min.   :0.84   Min.   : 0.400   Min.   :  22.0   Min.   : 0.000  
##  1st Qu.:2.29   1st Qu.: 8.607   1st Qu.: 310.0   1st Qu.: 1.000  
##  Median :2.70   Median :11.200   Median : 401.0   Median : 2.000  
##  Mean   :2.75   Mean   :11.819   Mean   : 415.4   Mean   : 3.189  
##  3rd Qu.:3.12   3rd Qu.:14.200   3rd Qu.: 498.0   3rd Qu.: 4.000  
##  Max.   :7.70   Max.   :49.800   Max.   :1185.0   Max.   :48.000  
##  NA&#39;s   :15     NA&#39;s   :14       NA&#39;s   :13       NA&#39;s   :29      
##     MCV.fl.           Lymp       
##  Min.   : 59.0   Min.   : 0.100  
##  1st Qu.: 81.0   1st Qu.: 3.570  
##  Median : 87.0   Median : 4.700  
##  Mean   : 87.1   Mean   : 5.186  
##  3rd Qu.: 94.0   3rd Qu.: 6.210  
##  Max.   :132.0   Max.   :22.600  
##  NA&#39;s   :14      NA&#39;s   :13</code></pre>
<p><strong>Categorigal variables</strong></p>
<pre><code>cm %&gt;% 
   select(SEX) %&gt;% 
   table() %&gt;% 
   as.data.frame() %&gt;% 
   na.omit() %&gt;%
   mutate(Prop = (Freq/sum(Freq))*100) -&gt; sex       # Sex

sex

##   . Freq  Prop
## 1 F  657 52.56
## 2 M  593 47.44

colnames(sex) &lt;- c(&quot;cat&quot;, &quot;freq&quot;, &quot;prop&quot;)
par(mar=c(4,4,3,4))
barplot(
   sex$prop, 
   ylim=c(0, 60), 
   col=c(2,4)
)
text(
   x=c(0.7,1.9), 
   y=30, 
   labels=c(&quot;Female&quot;, &quot;Male&quot;)
)
mtext(
   text = &quot;Proportion (%)&quot;,
   side = 2,
   line = 3
)
mtext(
   text = &quot;Sex&quot;,
   side = 1,
   line = 1
)</code></pre>
<img src="/home/kesoh/esoh/data/sadaccgwas/writeup/hbfgenomics/codebook/hbfgwascodebook_files/figure-markdown_strict/unnamed-chunk-5-1.png" alt="Figue 1. Proportion of females and males in the discovery dataset of Cameroonian inidividuals"  />
<p class="caption">
Figue 1. Proportion of females and males in the discovery dataset of Cameroonian inidividuals
</p>
<pre><code>cm %&gt;% select(Stroke) %&gt;% table()               # Number of patients with at least one overt stroke

## .
##   No  Yes 
## 1195   46

cm %&gt;% select(athal) %&gt;% table()        # 3.7 kb HbA1/HbA2 alpha thalassaemia deletion

## .
## α3.7/α3.7   αα/α3.7     αα/αα 
##        18       101       286

cm %&gt;% select(Haplotypes) %&gt;% table()       # HBB gene cluster haplotypes

## .
##     Atypical          BEN       BEN/AI BEN/Atypical      BEN/CAM      BEN/CAR 
##            3          331            2           26           99            2 
##          CAM       CAM/AI CAM/Atypical      CAM/SEN          CAR CAR/Atypical 
##          174            1            8            1            1            1 
##      CAR/CAM          SEN 
##            1            4</code></pre>
<p><em>CAM = CAM/CAM, BEN = BEN/BEN, SEN = SEN/SEN, CAR = CAR/CAR, Atypical = Atypical/Atypical</em></p>
<h2 id="summary-of-tanzania-data">Summary of Tanzania data</h2>
<h3 id="variables-1">Variables</h3>
<pre><code>tz %&gt;% names()

## [1] &quot;FID&quot; &quot;IID&quot; &quot;AGE&quot; &quot;HbF&quot; &quot;SEX&quot;</code></pre>
<h3 id="summary-statistics-1">Summary statistics</h3>
<p><strong>Quantitative/Continuous variables</strong></p>
<pre><code>variables &lt;- c(
   &quot;AGE&quot;,
   &quot;HbF&quot;
)
tz %&gt;% select(variables) %&gt;% summary()

##       AGE              HbF        
##  Min.   : 5.021   Min.   : 0.200  
##  1st Qu.: 7.876   1st Qu.: 2.600  
##  Median :11.069   Median : 4.800  
##  Mean   :13.288   Mean   : 5.748  
##  3rd Qu.:16.298   3rd Qu.: 7.800  
##  Max.   :44.310   Max.   :32.500</code></pre>
<p><strong>Categorical variable</strong></p>
<pre><code>tz %&gt;%
   mutate(sex.updated = ifelse(SEX == 1, &quot;M&quot;, ifelse(SEX == 2, &quot;F&quot;, NA))) %&gt;%
   select(sex.updated) %&gt;%
   table() %&gt;%
   as.data.frame() %&gt;%
   na.omit() %&gt;%
   mutate(Prop = (Freq/sum(Freq))*100) -&gt; sex           # Sex

sex

##   . Freq   Prop
## 1 F  539 52.895
## 2 M  480 47.105

colnames(sex) &lt;- c(&quot;cat&quot;, &quot;freq&quot;, &quot;prop&quot;)
par(mar=c(4,4,3,4))
barplot(
   sex$prop,
   ylim=c(0, 60),
   col=c(2,4)
)
text(
   x=c(0.7,1.9),
   y=30,
   labels=c(&quot;Female&quot;, &quot;Male&quot;)
)
mtext(
   text = &quot;Proportion (%)&quot;,
   side = 2,
   line = 3
)
mtext(
   text = &quot;Sex&quot;,
   side = 1,
   line = 1
)</code></pre>
<img src="/home/kesoh/esoh/data/sadaccgwas/writeup/hbfgenomics/codebook/hbfgwascodebook_files/figure-markdown_strict/unnamed-chunk-9-1.png" alt="Figure 2. Proportion of females and males in the replication dataset of Tanzanian inidividuals"  />
<p class="caption">
Figure 2. Proportion of females and males in the replication dataset of Tanzanian inidividuals
</p>
<hr />
<h1 id="hbf-normalization-cubic-root">HbF normalization (Cubic root)</h1>
<h3 id="cameroon">Cameroon</h3>
<pre><code>cm %&gt;% 
   mutate(normalizedHbF = HbF^(1/3)) -&gt; cm.hbf.norm 
   
write.table(
   cm.hbf.norm, 
   &quot;../clinicaldata/cm_clinical_data_updated.tsv&quot;,
   col.names = T,
   row.names = F,
   quote = F,
   sep = &quot;\t&quot;
)

cm.hbf.norm %&gt;% 
   select(HbF, normalizedHbF) %&gt;% 
   head()

## # A tibble: 6 × 2
##     HbF normalizedHbF
##   &lt;dbl&gt;         &lt;dbl&gt;
## 1  31.6          3.16
## 2   6.4          1.86
## 3  12.6          2.33
## 4   8.8          2.06
## 5  17.5          2.60
## 6  17.8          2.61</code></pre>
<h3 id="tanzania">Tanzania</h3>
<pre><code># Tanzania
tz %&gt;%
   mutate(normalizedHbF = HbF^(1/3)) -&gt; tz.hbf.norm

write.table(
   tz.hbf.norm,
   &quot;../clinicaldata/tz_clinical_data_updated.tsv&quot;,
   col.names = T,
   row.names = F,
   quote = F,
   sep = &quot;\t&quot;
)

tz.hbf.norm %&gt;% 
   select(HbF, normalizedHbF) %&gt;% 
   head()

## # A tibble: 6 × 2
##     HbF normalizedHbF
##   &lt;dbl&gt;         &lt;dbl&gt;
## 1   0.2         0.585
## 2   0.3         0.669
## 3   0.3         0.669
## 4   0.3         0.669
## 5   0.3         0.669
## 6   0.4         0.737</code></pre>
<h3 id="hbf-distribution-before-and-after-normalization">HbF distribution before and after normalization</h3>
<pre><code>cm.hbf.den &lt;- density(cm.hbf.norm$HbF)
cm.nhbf.den &lt;- density(cm.hbf.norm$normalizedHbF)
tz.hbf.den &lt;- density(tz.hbf.norm$HbF)
tz.nhbf.den &lt;- density(tz.hbf.norm$normalizedHbF)

#par(mfrow=c(2,2))
par(mar=c(4,4,3,4))
plot(cm.hbf.den, main=&quot;Cameroon HbF&quot;)
polygon(cm.hbf.den, col=2)
plot(cm.nhbf.den, ylab=&quot;&quot;, main=&quot;Cameroon normalized HbF&quot;)
polygon(cm.nhbf.den, col=5)
plot(tz.hbf.den, main=&quot;Tanzania HbF&quot;)
polygon(tz.hbf.den, col=2)
plot(tz.nhbf.den, ylab=&quot;&quot;, main=&quot;Tanzania normalized HbF&quot;)
polygon(tz.nhbf.den, col=5)</code></pre>
<img src="/home/kesoh/esoh/data/sadaccgwas/writeup/hbfgenomics/codebook/hbfgwascodebook_files/figure-markdown_strict/unnamed-chunk-12-1.png" alt="Figure 3. HbF distribution in Cameroonian and Tanzanian study participants before and after normalization" width="50%" /><img src="/home/kesoh/esoh/data/sadaccgwas/writeup/hbfgenomics/codebook/hbfgwascodebook_files/figure-markdown_strict/unnamed-chunk-12-2.png" alt="Figure 3. HbF distribution in Cameroonian and Tanzanian study participants before and after normalization" width="50%" /><img src="/home/kesoh/esoh/data/sadaccgwas/writeup/hbfgenomics/codebook/hbfgwascodebook_files/figure-markdown_strict/unnamed-chunk-12-3.png" alt="Figure 3. HbF distribution in Cameroonian and Tanzanian study participants before and after normalization" width="50%" /><img src="/home/kesoh/esoh/data/sadaccgwas/writeup/hbfgenomics/codebook/hbfgwascodebook_files/figure-markdown_strict/unnamed-chunk-12-4.png" alt="Figure 3. HbF distribution in Cameroonian and Tanzanian study participants before and after normalization" width="50%" />
<p class="caption">
Figure 3. HbF distribution in Cameroonian and Tanzanian study participants before and after normalization
</p>
